ctapipe/io/datawriter.py

```diff
@@ -500,6 +515,40 @@ def fill_from_simtel(
                         containers=hist_container,
                     )
 
+    def write_muon_events(self, event: ArrayEventContainer):
+        """
+        add muon entries to the event/telescope tables for each muon event
+        """
+        # perform delayed initialization on first event
+        if self._writer is None:
```

Comment:
this shouldn't be needed anymore (and will always be False), so just remove the two lines 522ff